WcaOnRails/app/webpacker/components/Media/MediaTable.jsx

```diff
@@ -0,0 +1,119 @@
+import _ from 'lodash'
+import React, { useState, useEffect } from 'react';
+import { Table } from "semantic-ui-react";
+import { countries } from '../../lib/wca-data.js.erb';
+import { competitionUrl } from '../../lib/requests/routes.js.erb';
+import moment from 'moment';
+
+function dateRangeBetween(fromDate, toDate) {
+    const format = new Intl.DateTimeFormat("en", {
+        year: "numeric",
+        month: "short",
+        day: "numeric",
+    });
+    const date1 = new Date(fromDate);
+    const date2 = new Date(toDate);
+    const ans = format.formatRange(date1, date2);
+    return ans;
+}
+
+function exampleReducer(state, action) {
+    switch (action.type) {
+        case 'CHANGE_SORT':
+            if (state.column === action.column) {
+                return {
+                    ...state,
+                    data: state.data.slice().reverse(),
```

Comment:
Unless this will be manipulated further (adding / removing items from this list), this can very likely be switched to a derived prop and we just compute it's state in the `MediaTable` component wrapping it with `useMemo`